pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/SegmentDeletionManager.java

```diff
@@ -166,6 +177,14 @@ protected synchronized void deleteSegmentFromPropertyStoreAndLocal(String tableN
         propStorePathList.add(segmentPropertyStorePath);
       }
 
+      for (SegmentDeletionListener segmentDeletionListener : _segmentDeletionListeners) {
+        try {
+          segmentDeletionListener.onSegmentDeletion(tableName, segmentsToDelete);
```

Comment:
Does it make sense to change it to a Collection in the interface?

Also, it may be useful to have other data along with segment name (start/end time, or creation time, etc.). So, does it make sense to make this a list of descriptors? Or, a list of segment metadata?